src/tainting/Dataflow_tainting.ml

```diff
@@ -1727,7 +1727,14 @@ let findings_from_arg_updates_at_exit enter_env exit_env : T.finding list =
   (* TOOD: We need to get a map of `lval` to `Taint.arg`, and if an extension
    * of `lval` has new taints, then we can compute its correspoding `Taint.arg`
    * extension and generate an `ArgToArg` finding too. *)
-  enter_env |> Lval_env.seq_of_tainted |> List.of_seq
+  enter_env |> Lval_env.seq_of_tainted
+  |> Seq.flat_map (fun ({ base; _ }, enter_taints) ->
+         (* We need to consider all lvals of the same base component
+            due to index sensitivity. *)
```

Comment:
due to **field and** index sensitivity ?

Reply:
done.